app/templates/components/template-filter.html

```diff
@@ -0,0 +1,130 @@
+{# 
+    This macro creates a filterable template list based on specified criteria. 
+    The macro requires three parameters:
+        - row_selector: A CSS selector for the rows to filter, i.e '.template-row'
+        - notification_types: list of notification types, i.e ['Email', 'SMS']
+        - template_categories: list of template categories and text to display, i.e ['Status Update', 'Password Reset']  
+        - notification_type_data_attribue: The data attribute to filter by notification type, i.e 'data-notification-type'  
+        - template_category_data_attribute: The data attribute to filter by template category, i.e 'data-template-category'
+    
+    NOTE: the data attributes need to be present on the rows indicated by `row-selector` for the filtering to work.  Example
+    of a complete row:
+    ```
+    <div class="template-row" data-notification-type="email" data-template-category="status-update">
+    ```
+#}
+{% macro template_filter(row_selector, notification_types, template_categories, notification_type_data_attribue, template_category_data_attribute) %}
+    {# Ensure all required parameters are provided #}
+    {% if not row_selector or not notification_type_data_attribue or not template_category_data_attribute %}
+        <div class="text-red font-bold my-4">
+            Missing required parameters: <code>row_selector</code>, <code>notification_type_attribute</code>, and <code>template_category_attribute</code>. Please verify your code and try again.
+        </div>
+    {% else %}
+        {# Main filter container with data attributes for dynamic filtering #}
+        <div class="template-filter" data-row-selector="{{ row_selector }}">
+            <details data-testid="filter">
+                <summary data-testid="filter-toggle">{{ _("Apply filters") }}</summary>
+                <nav class="flex p-0 gap-gutter pl-gutter mt-2" aria-label="{{ _('Filter by template type and category') }}" data-testid="filter-content">
```

Comment:
We can remove pl gutter here to keep things aligned. 

```suggestion
                <nav class="flex p-0 gap-gutter mt-2" aria-label="{{ _('Filter by template type and category') }}" data-testid="filter-content">
```